File: WebClient/src/app/search/dtos/global-search-result.dto.ts
import { CommunitySummaryDto } from "../../communities/dtos/community-summary.dto";
import { JobSummaryDto } from "../../jobs/dtos/job-summary.dto";
import { PostSummaryDto } from "../../content/posts/dtos/post-summary.dto";
import { ProjectSummaryDto } from "../../projects/dtos/project-summary.dto";
import { ProfileSummaryDto } from "../../social-graph/dtos/profile-summary.dto";
import { ProblemSummaryDto } from "../../qa/dtos/problem-summary-dto";

export interface GlobalSearchResultDto {
    profiles: ProfileSummaryDto[];
    jobs: JobSummaryDto[];
    communities: CommunitySummaryDto[];
    posts: PostSummaryDto[];
    projects: ProjectSummaryDto[];
    questions: ProblemSummaryDto[];
}

File: backend/SNS.Application/Search/Queries/GetGlobalSearch/GlobalSearchResultDto.cs
using SNS.Application.ContentManagement.Communities.Communities.Contracts;
using SNS.Application.ContentManagement.Posts.Posts.Contracts;
using SNS.Application.Discussions.Problems.Problems.Contracts;
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Application.Projects.Contracts;
using SNS.Application.Search.Jobs.Contracts;

namespace SNS.Application.Search.Queries.GetGlobalSearch;

/// <summary>
/// Represents response DTO for global search containing matched top results grouped across multiple entity categories.
/// </summary>
public sealed record GlobalSearchResultDto
{
    /// <summary>
    /// Gets matched user profile summaries.
    /// </summary>
    public List<ProfileSummaryDto> Profiles { get; init; } = new();

    /// <summary>
    /// Gets matched project overviews.
    /// </summary>
    public List<ProjectOverviewDto> Projects { get; init; } = new();

    /// <summary>
    /// Gets matched community summaries.
    /// </summary>
    public List<CommunitySummaryDto> Communities { get; init; } = new();

    /// <summary>
    /// Gets matched job summaries.
    /// </summary>
    public List<JobSummaryDto> Jobs { get; init; } = new();

    /// <summary>
    /// Gets matched discussion problem summaries.
    /// </summary>
    public List<ProblemSummaryDto> Problems { get; init; } = new();

    /// <summary>
    /// Gets matched post overviews.
    /// </summary>
    public List<PostOverviewDto> Posts { get; init; } = new();
}
